src/SFA.DAS.EAS.Portal/Application/Commands/Cohort/CohortApprovalRequestedCommandHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using SFA.DAS.Commitments.Api.Client.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using System;
+using SFA.DAS.EAS.Portal.Application.Services;
+using SFA.DAS.EAS.Portal.Client.Types;
+
+namespace SFA.DAS.EAS.Portal.Application.Commands.Cohort
+{
+    public class CohortApprovalRequestedCommandHandler : ICommandHandler<CohortApprovalRequestedCommand>
+    {
+        private readonly IAccountDocumentService _accountsService;
+        private readonly IProviderCommitmentsApi _providerCommitmentsApi;
+
+        public CohortApprovalRequestedCommandHandler(
+            IAccountDocumentService accountsService, 
+            IProviderCommitmentsApi providerCommitmentsApi)
+        {
+            _accountsService = accountsService;
+            _providerCommitmentsApi = providerCommitmentsApi;
+        }
+
+        public async Task Handle(CohortApprovalRequestedCommand command, CancellationToken cancellationToken = default)
+        {
+            var accountDocumentTask = _accountsService.Get(command.AccountId, cancellationToken);
+            var commitmentTask = _providerCommitmentsApi.GetProviderCommitment(command.ProviderId, command.CommitmentId);
+            
+            var accountDocument = await accountDocumentTask;
+            var account = accountDocument.Account;
+            var commitment = await commitmentTask;
+
+            var cohortReference = commitment.Reference;            
+            // TODO: there is an assumption there is only one Organisation (which is true for the initial development) but in the furture may not be true based on the events we will receive.
```

Comment:
The requirement for MVS is to only display a single cohort approval request in the front end. We should still support handling multiple events and correctly storing the information in our store.

If we leave this undone for the release of this PR, we'll need to write a web job (or similar), to correct the stored data. We'll have to enumerate all cohorts in all account documents, call the commitments api to refetch the legal entity id, then move the cohort to the correct account legal entity. That task to create the throwaway webjob will be of a similar scope to the original story (if not larger) of consuming and storing the `CohortApprovalRequestedByProvider`. Much better to write the handful of lines to do the right thing as part of this PR.